src/optimizer_reader.c

```diff
@@ -224,16 +224,16 @@ IndexIterator *NewOptimizerIterator(QOptimizer *qOpt, IndexIterator *root, Itera
   oi->numDocs = qOpt->sctx->spec->docs.size;
   oi->childEstimate = root->NumEstimated(root->ctx);
 
+  const FieldSpec *field = IndexSpec_GetField(qOpt->sctx->spec, qOpt->fieldName);
```

Comment:
`NULL` case handling missing

Reply:
It was missing before, so we have the same behavior.
I assume we make the validation before, so now we know it will succeed.
We might need to investigate this further